Temperature_API/router.py

```diff
@@ -0,0 +1,46 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from . import schemas, crud
+from dependencies import get_db
+from .schemas import TemperatureList
+
+router = APIRouter()
+
+
+# Get all temperature records
+@router.get("/temperatures/", response_model=list[schemas.TemperatureList])
+async def get_all_temperatures(db: AsyncSession = Depends(get_db)):
+    """
+    Get a list of all temperature records.
+    """
+    return await crud.get_all_temperatures(db=db)
```

Comment:
Consider adding error handling for database connection issues or unexpected errors when retrieving all temperature records. This will help in providing more informative responses to the client.